java-squid/src/main/java/org/sonar/java/resolve/JavaType.java

```diff
@@ -228,6 +230,9 @@ public boolean isSubtypeOf(String fullyQualifiedName) {
 
     @Override
     public boolean isSubtypeOf(Type superType) {
+      if (superType == Symbols.anyType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)